src/wlserver.cpp

```diff
@@ -187,6 +187,15 @@ static void wlserver_handle_pointer_button(struct wl_listener *listener, void *d
 	wlr_seat_pointer_notify_frame( wlserver.wlr.seat );
 }
 
+static void wlserver_handle_pointer_axis(struct wl_listener *listener, void *data)
+{
+	struct wlserver_pointer *pointer = wl_container_of( listener, pointer, axis );
+	struct wlr_event_pointer_axis *event = (struct wlr_event_pointer_axis *) data;
+
+	wlr_seat_pointer_notify_axis( wlserver.wlr.seat, event->time_msec, event->orientation, event->delta, event->delta_discrete, event->source );
+	wlr_seat_pointer_notify_frame( wlserver.wlr.seat );
```

Comment:
Instead of sending `frame` manually, we should listen for `device->pointer->events.frame`.

Reply:
Ah, this version of wlroots might be too old for this. Nevermind!

Reply:
Hm, it seems like it's not too old after all!

Reply:
OK, I will try that.

Reply:
I have changed it.

Reply:
I didn't realize our `motion` event handlers weren't doing this properly.